api/src/main/java/org/openmrs/validator/PersonNameValidator.java

```diff
@@ -129,11 +129,26 @@ public void validatePersonName(PersonName personName, Errors errors, boolean arr
 			return;
 		}
 		// Make sure they assign a name
-		if (StringUtils.isBlank(personName.getGivenName())
+		if (StringUtils.isBlank(personName.getGivenName()) 
 		        || StringUtils.isBlank(personName.getGivenName().replaceAll("\"", ""))) {
 			errors.rejectValue(getFieldKey("givenName", arrayInd, testInd), "Patient.names.required.given.family");
 		}
 
+		// Make sure entered name value has no leading and trailing spaces
+		if (StringUtils.isNotBlank(personName.getGivenName()) && startsOrEndsWithWhitespace(personName.getGivenName())) {
```

Comment:
please move `StringUtils.isNotBlank(personName.getGivenName())` into the private method so we dont have to repeat this for every field, plus if its blank it does not start or end with whitespace so this fits into the duty of the method